dashboard/dashboard.py

```diff
@@ -1,43 +1,229 @@
+        encoder_depth=cfg.model.encoder_depth,
+        encoder_num_heads=cfg.model.encoder_num_heads,
+        decoder_dim=cfg.model.decoder_dim,
+        decoder_depth=cfg.model.decoder_depth,
+        decoder_num_heads=cfg.model.decoder_num_heads,
+        mlp_ratio=cfg.model.mlp_ratio,
+    ).to(device)
 
-    with open("tmp/processed_files.pickle", "rb") as handle:
-        unserialized_data = pickle.load(handle)
+    model.load_state_dict(checkpoint["model"])
+    model.eval()
 
-    files = unserialized_data.keys()
+    return model
 
-    selected_filename = st.selectbox("Select piece to display", options=files)
 
-    display_pianoroll(
-        title=selected_filename,
-        midi_pieces=unserialized_data[selected_filename],
+def model_selection() -> MidiMaskedAutoencoder:
+    checkpoints = glob("checkpoints/*")
+    checkpoint_path = st.selectbox("Select checkpoint", options=checkpoints)
+
+    model = get_model(checkpoint_path)
+    return model
+
+
+@st.cache_data
+def get_dataset() -> Dataset:
+    dataset_name = "JasiekKaczmarczyk/maestro-v1-sustain-masked"
+    dataset = load_dataset(dataset_name, split="validation")
+    return dataset
+
+
+def main():
+    # Load model
+    model = model_selection()
+
+    # Prep data
+    dataset = get_dataset()
+    source = [json.loads(source) for source in dataset["source"]]
+    source_df = pd.DataFrame(source)
+
+    composers = source_df.composer.unique()
+    selected_composer = st.selectbox(
+        label="Select composer",
+        options=composers,
+        index=3,
+    )
+
+    ids = source_df.composer == selected_composer
+    piece_titles = source_df[ids].title.unique()
+    selected_title = st.selectbox(
+        label="Select title",
+        options=piece_titles,
     )
+    st.write(selected_title)
+
+    ids = (source_df.composer == selected_composer) & (source_df.title == selected_title)
+    n_samples = 10
+    seed = 137
+    part_df = source_df[ids].sample(n_samples, random_state=seed)
+
+    idxs = part_df.index.values
+    part_dataset = dataset.select(idxs)
+    midi_dataset = MidiDataset(part_dataset)
+    generated_pieces = generate_pieces(midi_dataset=midi_dataset, model=model)
+
+    for processing_result in generated_pieces:
+        display_pianoroll(processing_result)
+
+
+@torch.no_grad()
+def generate_pieces(
+    midi_dataset: MidiDataset,
+    model: MidiMaskedAutoencoder,
+):
+    dataloader = DataLoader(midi_dataset, batch_size=256, shuffle=True)
+
+    generated_pieces = []
+    for batch in dataloader:
+        pitches = batch["pitch"].to(device)
+        velocities = batch["velocity"].to(device)
+        dstarts = batch["dstart"].to(device)
+        durations = batch["duration"].to(device)
+
+        pred_pitches, pred_velocities, pred_dstarts, pred_durations, mask = model(
+            pitch=pitches,
+            velocity=velocities,
+            dstart=dstarts,
+            duration=durations,
+            masking_ratio=0.5,
```

Comment:
Since we generate pieces on the fly, we could add ```masking_ratio``` as parameter for adjusting with some kind of slider

Reply:
done!